File: frontend/src/services/TimeSlotService.ts
import axios from "axios";

export default class TimeSlotService {
    getTimeSlotsByDate(doctorId:string, slotDate:string) {
        return axios.get(`/api/time-slots/doctor/${doctorId}/${slotDate}`)
    }

    addPatientId(slotId:string, patientId:string) {
        return axios.put(`/api/time-slots/patient/${slotId}/${patientId}`)
    }

    removePatientId(slotId:string) {
        return axios.put(`/api/time-slots/patient-remove/${slotId}`)
    }

    getTimeSlotById(id:string) {
        return axios.get(`/api/time-slots/${id}`)
    }

    getTimeSlotByVerificationCode(contact:string, verificationCode:string) {
        return axios.get(`/api/time-slots/verification-code/${contact}/${verificationCode}`)
    }
}

File: backend/src/main/java/de/earzt/backend/services/abstracts/TimeSlotService.java
package de.earzt.backend.services.abstracts;

import de.earzt.backend.services.dtos.requests.TimeSlotMultiRequest;
import de.earzt.backend.services.dtos.responses.TimeSlotCreatedResponse;
import de.earzt.backend.services.dtos.responses.TimeSlotGetAllResponse;

import java.time.LocalDate;
import java.util.List;

public interface TimeSlotService {
    List<TimeSlotGetAllResponse> getTimeSlotsByDate(LocalDate slotDate, String doctorId);
    List<TimeSlotGetAllResponse> addMultiTimeSlot(TimeSlotMultiRequest timeSlotMultiRequest);
    TimeSlotCreatedResponse getTimeSlotById(String id);
    TimeSlotCreatedResponse changeTimeSlotStatus(String id);
    TimeSlotCreatedResponse addPatientId(String slotId, String patientId);
}
